lib/DBDish.pm6

```diff
@@ -19,6 +19,17 @@ role Driver does DBDish::ErrorHandling {
     #method new() { nextsame; }
 }
 
+role Type {
+    has Sub %!Conversions{Str};
+
+    method set(Str $name, Sub $convert) {
```

Comment:
I'd just adjust the ```$convert``` to ```&convert``` and lose the explicit type constraint personally (that way the caller is free to implement it how they want as long as it supports Callable semantics.)